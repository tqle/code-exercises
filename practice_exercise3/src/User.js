import React from "react";
import { Link } from "@reach/router";

import FormattedAddress from "./FormattedAddress";
import EmailLink from "./EmailLink";

class User extends React.Component {
  render() {
    // Destructuring objects from props.
    const { id, name, username, email, address } = this.props;
    // Check address fields, cannot assume address is an object in all cases.
    const street = address && address.street;
    const suite = address && address.suite;
    const state = address && address.state;
    const city = address && address.city;
    const zipCode = address && address.zipCode;
    return (
      <div>
        {/* Router link to details page */}
        <Link to={`/details/${id}`}>
          <h1>{name}</h1>
        </Link>
        <div>{username}</div>
        <EmailLink email={email} />
        <FormattedAddress
          street={street}
          suite={suite}
          city={city}
          state={state}
          zipCode={zipCode}
        />
      </div>
    );
  }
}

export default User;
