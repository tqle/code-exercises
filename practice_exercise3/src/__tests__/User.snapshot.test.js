import React from "react";
import { create } from "react-test-renderer";
import User from "../User";

/**
 * Test User snapshot
 */
test("<User /> renders correctly with address object", () => {
  const id = "12345";
  const email = "johntest@sometest.com";
  const name = "John Test";
  const userName = "johnTest";

  const address = {
    street: "Kulas Light",
    suite: "Apt. 556",
    city: "Gwenborough",
    zipcode: "92998-3874"
  };

  const test = create(
    <User
      id={id}
      name={name}
      username={userName}
      email={email}
      address={address}
    />
  );
  expect(test.toJSON()).toMatchSnapshot();
});

test("<User /> renders correctly with address string", () => {
  const id = "12345";
  const email = "johntest@sometest.com";
  const name = "John Test";
  const userName = "johnTest";

  const address = {
    street: "Kulas Light",
    suite: "Apt. 556",
    city: "Gwenborough",
    zipcode: "92998-3874"
  };

  const test = create(
    <User
      id={id}
      name={name}
      username={userName}
      email={email}
      address={address}
    />
  );
  expect(test.toJSON()).toMatchSnapshot();
});
