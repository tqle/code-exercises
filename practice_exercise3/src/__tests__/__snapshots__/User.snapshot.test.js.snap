// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> renders correctly with address object 1`] = `
<div>
  <a
    href="/details/12345"
    onClick={[Function]}
  >
    <h1>
      John Test
    </h1>
  </a>
  <div>
    johnTest
  </div>
  <a
    href="mailto:johntest@sometest.com"
    title="johntest@sometest.com"
  >
    johntest@sometest.com
  </a>
  <address
    className="normal-style"
  >
    Kulas Light
     
    Apt. 556
     
    <br />
    Gwenborough
    , 
    FL
    , 
    
  </address>
</div>
`;

exports[`<User /> renders correctly with address string 1`] = `
<div>
  <a
    href="/details/12345"
    onClick={[Function]}
  >
    <h1>
      John Test
    </h1>
  </a>
  <div>
    johnTest
  </div>
  <a
    href="mailto:johntest@sometest.com"
    title="johntest@sometest.com"
  >
    johntest@sometest.com
  </a>
  <address
    className="normal-style"
  >
    Kulas Light
     
    Apt. 556
     
    <br />
    Gwenborough
    , 
    FL
    , 
    
  </address>
</div>
`;
